feat: Implement Input component

choiseona opened the pull request:
## 📂 작업 내용

closes #12 

- [x]  Default 컴포넌트 구현
- [x]  props로 placeholder, label 받기

## 💡 자세한 설명

placeholder와 label을 props로 받는, 재사용 가능한 Input 컴포넌트 구현

## 📗 참고 자료 & 구현 결과 (선택)

![image](https://github.com/user-attachments/assets/480104a2-a13b-4ac8-9cae-7f65d941a3d6)
![image](https://github.com/user-attachments/assets/c2814ed9-5415-4002-b763-1f112245b8df)

## ✅ 셀프 체크리스트

- [x] PR 제목을 형식에 맞게 작성했나요?
- [x] 브랜치 전략에 맞는 브랜치에 PR을 올리고 있나요? (`main`이 아닙니다.)
- [x] 이슈는 close 했나요?
- [x] Reviewers, Labels를 등록했나요?
- [x] 작업 도중 문서 수정이 필요한 경우 잘 수정했나요?
- [x] 불필요한 코드는 제거했나요?

choiseona committed:
feat: Add reusable Input component

- Implement reusable Input component with class-variance-authority
- Add default variant with violet theme styling
- Include accessibility features with proper label
- Support forwarded refs and custom className props

kwaksj329 commented on client/src/components/ui/Input.tsx:
```diff
@@ -0,0 +1,48 @@
+import * as React from 'react';
```
리액트 17버전부터는 `import React`문 사용하지 않아도 됩니다!
아래 `React.InputHTMLAttributes`, `React.forwardRef`, `React.useId()` 만 정리해주시면 좋을 것 같아요. 👍🏻

choiseona replied:
수정했습니다!!

choiseona committed:
refactor: Replace React wildcard import with named imports

- Import forwardRef and useId explicitly
- Remove unnecessary React namespace import

choiseona committed:
refactor: replace React wildcard import with named imports

- Import forwardRef and useId and InputHTMLAttributes explicitly
- Remove unnecessary React namespace import

rhino-ty commented on client/src/components/ui/Input.tsx:
```diff
@@ -0,0 +1,48 @@
+import { cva, type VariantProps } from 'class-variance-authority';
+import { cn } from '@/utils/cn';
+
+const inputVariants = cva('px-4', {
```
사실 `variants`의 하위 객체들이 1개 밖에 없다면 그냥 직접 넣어주는 것도 괜찮습니다. 이런 식이면 조금 오버엔지니어링에 가까워서,, 그외 `forwardRef` 등도 그렇구요.
나중에 확장성을 위해 이렇게 했다면 그대로 진행해도 되구요!

choiseona replied:
아래와 같이 불필요한 것들 삭제해주었습니다!! 

```javascript
const Input = ({ className, label, ...props }: InputProps) => {
  const inputId = useId();

  return (
    <>
      <label htmlFor={inputId} className="block text-violet-950">
        {label}
      </label>
      <input
        id={inputId}
        type="text"
        className={cn(
          'h-11 w-full rounded-lg border-2 border-violet-950 px-4 text-base text-violet-950 placeholder:text-eastbay-500',
          className,
        )}
        {...props}
      />
    </>
  );
};
```

rhino-ty commented on client/src/components/ui/Input.tsx:
```diff
@@ -0,0 +1,48 @@
+export interface InputProps
+  extends Omit<InputHTMLAttributes<HTMLInputElement>, 'size'>,
+    VariantProps<typeof inputVariants> {
+  asChild?: boolean;
```
`asChild`는 사용하지 않으니 그냥 없애도 됩니다.

rhino-ty reviewed (approved):
수고하셨습니다! Approve는 하는데 제가 리뷰 단 것들 조금 생각해보시고 반영이나 아니면 그대로 머지 해주세요!

choiseona committed:
refactor: remove ref and forwardRef, simplify styling

- Converted Input to a regular functional component by removing unnecessary ref and forwardRef
- Applied styles directly within the className prop as a single design system is in use
